byway/src/main/webapp/recommendations.js

```diff
@@ -0,0 +1,195 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global google */
+/* global destination:writable, step:writable, interest:writable, region:writable, result:writable */
+
+if(document.readyState === 'loading') {
+  document.addEventListener('DOMContentLoaded', initialize);
+} else {
+  initialize();
+}
+
+let map;
+let placesService;
+
+// Measured in meters
+let RADIUS = 1000;
+let MIN_DISTANCE = 4000;
+
+let interests = ["park"];
+let destinations = new Set();
+let regions = [];
+
+/**
+ * Initializes the webpage with a map and other google
+ * services. Creates a route between two endpoints.
+ */
+function initialize() {
+  // Modified version of Justine's implementation
+  var directionsService = new google.maps.DirectionsService();
+  var directionsRenderer = new google.maps.DirectionsRenderer();
+  var start = new google.maps.LatLng(37.7699298, -122.4469157);
+  var end = new google.maps.LatLng(37.7683909618184, -122.51089453697205);
+  var mapOptions = {
+    zoom: 14,
+    center: start
+  }
+  addDestinations(start, end);
+  map = new google.maps.Map(document.getElementById('map'), mapOptions);
+  directionsRenderer.setMap(map);
+  document.getElementById("route").addEventListener("click", () => {
+    calcRoute(directionsService, directionsRenderer, start, end);
+  });
+  placesService = new google.maps.places.PlacesService(map);
+}
+
+/**
+ * Add start and end points of the route to regions to search around.
+ * Temporary set up to mimic adding several endpoints between legs.
+ */
+function addDestinations(start, end) {
+  destinations.add(start);
+  destinations.add(end);
+  for(destination of destinations) {
+    regions.push(destination);
```

Comment:
Can use the spread operator instead: ``` regions.push(...destinations) ```. More example usage can be found at: https://www.javascripttutorial.net/es6/javascript-spread/

Reply:
Updated